src/view/writer.js

```diff
@@ -399,6 +412,68 @@ export function remove( range ) {
 }
 
 /**
+ * Removes matching elements from given range.
+ *
+ * Throws {@link module:utils/ckeditorerror~CKEditorError CKEditorError} `view-writer-invalid-range-container` when
+ * {@link module:engine/view/range~Range#start start} and {@link module:engine/view/range~Range#end end} positions are not placed inside
+ * same parent container.
+ *
+ * @function module:engine/view/writer~writer.clear
+ * @param {module:engine/view/range~Range} range Range to clear.
+ * @param {module:engine/view/element~Element} element Element to remove.
+ */
+export function clear( range, element ) {
+	validateRangeContainer( range );
+
+	// Create walker on given range.
+	// We walk backward because when we remove element during walk it modify range end position.
+	const walker = range.getWalker( {
+		direction: 'backward',
+		ignoreElementEnd: true
+	} );
+
+	// Let's walk.
+	for ( const current of walker ) {
+		const item = current.item;
+		let rangeToRemove;
+
+		// When current item matches to the given element.
+		if ( item instanceof Element && element.isSimilar( item ) ) {
+			// Create range on this element.
+			rangeToRemove = Range.createOn( item );
+		// When range starts inside Text or TextProxy element.
+		} else if ( !current.nextPosition.isAfter( range.start ) && ( item instanceof Text || item instanceof TextProxy ) ) {
+			// We need to check if parent of this text matches to given element.
+			const parentElement = item.getAncestors().find( ( ancestor ) => {
+				return ancestor instanceof Element && element.isSimilar( ancestor );
+			} );
+
+			// If it is then create range inside this element.
+			if ( parentElement && parentElement ) {
+				rangeToRemove = Range.createIn( parentElement );
+			}
+		}
+
+		// If we have found element to remove.
+		if ( rangeToRemove ) {
+			// We need to check if element range stick out of the given range and truncate if it is.
+			if ( !range.containsRange( rangeToRemove ) ) {
```

Comment:
I mean, if `range` does contain `rangeToRemove`, neither of below conditions will be met and it is fine. You have no CPU gain because you have to perform `containsRange` check anyway, which also checks `isAfter` and `isBefore`. Additionally, this `if` makes the code less clean and understandable.

If you don't have any counterexample, check if any tests fail after removing this `if` and if not, remove it.